src/ResourceManager/IotCentral/Commands.IotCentral/IotCentral/NewAzureRmIotCentralApp.cs

```diff
@@ -106,6 +103,28 @@ public override void ExecuteCmdlet()
 
         }
 
+        private void CheckNameAvailability()
+        {
+            var checkNameInputs = new OperationInputs(this.Name, "IoTApps");
+            var nameAvailabilityInfo = this.IotCentralClient.Apps.CheckNameAvailability(checkNameInputs);
+            this.CheckAvailability(nameAvailabilityInfo);
+        }
+
+        private void CheckSubdomainAvailability()
+        {
+            var checkSubdomainInputs = new OperationInputs(this.Subdomain, "IoTApps");
+            var subdomainAvailabilityInfo = this.IotCentralClient.Apps.CheckSubdomainAvailability(checkSubdomainInputs);
+            this.CheckAvailability(subdomainAvailabilityInfo);
+        }
+
+        private void CheckAvailability(AppAvailabilityInfo availabilityInfo)
+        {
+            if (!(availabilityInfo.NameAvailable == true))
```

Comment:
Simplify this to
`if (availabilityInfo.NameAvailable != true)`

Reply:
@emgarten addressed all your reviews